File: scripts/final_live_smoke.py
#!/usr/bin/env python3
import os, sys, requests
base = os.getenv('API_BASE_URL', '').rstrip('/')
if not base:
    print('API_BASE_URL is verplicht', file=sys.stderr)
    sys.exit(1)
paths = ['/health', '/api/v1/health', '/api/v1/public/config']
failed = False
for path in paths:
    url = base + path
    try:
        r = requests.get(url, timeout=20)
        print(('OK  ' if r.ok else 'WARN'), r.status_code, url)
        if not r.ok:
            failed = True
    except Exception as exc:
        print('FAIL', url, str(exc), file=sys.stderr)
        failed = True
print('READY' if not failed else 'CHECK_FAILED')
sys.exit(1 if failed else 0)

File: scripts/final_live_smoke.mjs
#!/usr/bin/env node
const base = (process.env.MARKETING_BASE_URL || '').replace(/\/$/, '');
if (!base) {
  console.error('MARKETING_BASE_URL is verplicht');
  process.exit(1);
}
const targets = [
  '/api/public/config',
  '/app/login.html',
  '/app/set-password.html',
  '/api/onboarding/status',
  '/api/checkout/status?orderRef=phase4-smoke',
];
for (const path of targets) {
  const url = base + path;
  const res = await fetch(url, { redirect: 'manual', headers: { accept: 'application/json,text/html' } }).catch((error) => ({ ok: false, status: 0, error }));
  if (!res || res.status === 0) {
    console.error(`FAIL ${url}: ${res?.error?.message || 'request_failed'}`);
    process.exitCode = 1;
    continue;
  }
  console.log(`${res.ok ? 'OK  ' : 'WARN'} ${res.status} ${url}`);
}
